miner/worker.go

```diff
@@ -1000,26 +1012,23 @@ func (w *worker) fillTransactions(interruptCh chan int32, env *environment) {
 		}
 	}
 
-	var stopTimer *time.Timer
-	delay := w.engine.Delay(w.chain, env.header, &w.config.DelayLeftOver)
-	if delay != nil {
-		stopTimer = time.NewTimer(*delay)
-		log.Debug("Time left for mining work", "delay", delay.String())
-		defer stopTimer.Stop()
-	}
-
+	err = nil
 	if len(localTxs) > 0 {
 		txs := types.NewTransactionsByPriceAndNonce(env.signer, localTxs, env.header.BaseFee)
-		if w.commitTransactions(env, txs, interruptCh, stopTimer) {
+		err = w.commitTransactions(env, txs, interruptCh, stopTimer)
+		if err == errBlockInterruptedByNewHead || err == errBlockInterruptedByOutOfGas || err == errBlockInterruptedByTimeout {
```

Comment:
Please comment why `errBlockInterruptedByRecommit` can continue, otherwise it is confusing.